### ISSUE TITLE
gate failure cause all CI failed

### ISSUE THREAD

**Issue by jichenjc** [kubernetes/cloud-provider-openstack#2528]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

something wrong for all CI test like this 
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-openstack/2527/openstack-cloud-controller-manager-e2e-test/1750554290813931520

```
ovisioner-84db5d44d9-b7s5x    0/1     Pending             0          5m23s   <none>      <none>       <none>           <none>",
        "coredns-6799fbcd5-p7m2z                    0/1     Pending             0          5m23s   <none>      <none>       <none>           <none>",
        "openstack-cloud-controller-manager-rhpgn   0/1     ImageInspectError   0          2m24s   10.1.0.38   k3s-master   <none>           <none>",
```

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version:
- OpenStack version:
- Others:

**Comment by jichenjc** [comment]
https://github.com/k3s-io/k3s/issues/1857  seems has `invalid capacity 0 on image filesystem` as well 

and found the image is actually built , so need inspect what's the reason here

```
docker buildx build --output type=registry \
	--build-arg VERSION=v0.0.99 \
	--tag localhost/openstack-cloud-controller-manager:v0.0.99 \
	--platform linux/amd64 \
	--target openstack-cloud-controller-manager \
	.
```

**Comment by jichenjc** [comment]
I built image and run locally successfully ,so not sure what happened in CI gate ..
```
        image: 10.43.0.185/openstack-cloud-controller-manager:v0.0.99

```

**Comment by dulek** [comment]
I think the cause is Mirantis/cri-dockerd#299. Nevertheless, as mentioned on the ticket, nobody's testing K8s with Docker anymore and there's no point in us doing that. My fix in #2529 solves this by switching to containerd as a container runtime of the K8s we deploy in the CI.
